Mage.Tests/src/test/java/org/mage/test/cards/enchantments/ReadAheadTest.java

```diff
@@ -0,0 +1,71 @@
+package org.mage.test.cards.enchantments;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+/**
+ * @author TheElk801
+ */
+public class ReadAheadTest extends CardTestPlayerBase {
+
+    private static final String war = "The Elder Dragon War";
+
+    @Test
+    public void testElderDragonWarChapter1() {
+        addCard(Zone.HAND, playerA, war);
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain", 4);
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, war);
+        setChoice(playerA, "X=1");
+
+
+        setStopAt(1, PhaseStep.END_TURN);
+        setStrictChooseMode(true);
+        execute();
+
+        assertLife(playerB, 20 - 2);
+        assertPermanentCount(playerA, war, 1);
+        assertGraveyardCount(playerA, war, 0);
+    }
+
+    @Test
+    public void testElderDragonWarChapter2() {
+        addCard(Zone.HAND, playerA, war);
+        addCard(Zone.HAND, playerA, "Island");
+        addCard(Zone.BATTLEFIELD, playerA, "Mountain", 4);
+
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, war);
+        setChoice(playerA, "X=2");
+        setChoice(playerA, "Island");
+
+        setStopAt(1, PhaseStep.END_TURN);
+        setStrictChooseMode(true);
+        execute();
+
+        assertLife(playerB, 20);
+        assertGraveyardCount(playerA, "Island", 1);
+        assertHandCount(playerA, "Island", 0);
```

Comment:
Don't the tests start the players off with a library full of basic lands?  If so, this test could randomly fail if it happens to loot into another Island (or have one in the opening hand?).

Reply:
yeah that makes sense, I've changed it